### ISSUE TITLE
Documentation problem for gradient clipping parameter

### ISSUE THREAD

**Issue by isaacgerg** [tensorflow/tensorflow#52348]
This issue reflect documentation of this: https://www.tensorflow.org/api_docs/python/tf/keras/optimizers/Optimizer

As stated in the above documentation (relevant section c and pasted below), there is no difference between clipnorm and global_clipnorm.  However, looking at the code shows these parameters do 2 different things.  Recommend to update documentation.

clipnorm | float or None. If set, clips gradients to a maximum norm.
clipvalue | float or None. If set, clips gradients to a maximum value.
global_clipnorm | float or None. If set, clips gradients to a maximum norm.
iterations

**Comment by isaacgerg** [comment]
You didn't answer my question in my last response so I cannot move forward.

**Comment by tilakrayal** [comment]
@isaacgerg 
Can you please create the PR for the statement mentioned above.Thanks

**Comment by isaacgerg** [comment]
@tilakrayal Unfortunately, my employer does not allow me to submit PRs.

**Comment by sanatmpa1** [comment]
@isaacgerg,

clipnorm | float or None. If set, the gradient of each weight is individually clipped so that its norm is no higher than this value.
clipvalue | float or None. If set, the gradient of each weight is clipped to be no higher than this value.
global_clipnorm | float or None. If set, the gradient of all weights is clipped so that their global norm is no higher than this value

Do you think modifying the description as mentioned above would make it clear? I can submit a PR if that looks good. Thanks!

**Comment by isaacgerg** [comment]
@sanatmpa1 It is unclear to me if the description you list is correct.

A gradient is really a group of scalar values.  Its unclear how clipnorm differs from global_clipnorm.  My guess is that clipnorm groups weights by their function but its not clear to me.  Having an example to make this completely clear would be very helpful.  

Also, does clipvalue clip truly  by value only or by magnitude and preserve the sign?

**Comment by sachinprasadhs** [comment]
Let's consider the below optimizer,
```
optimizer = keras.optimizers.SGD(clipvalue=1.0)
model.compile(loss="mse", optimizer=optimizer)
```

This optimizer will clip every component of the gradient vector to a value between –1.0 and 1.0. This means that all the partial derivatives of the loss (with regard to each and every trainable parameter) will be clipped between –1.0 and 1.0. Note that it may change the orientation of the gradient vector. For instance, if the original gradient vector is [0.9, 100.0], it points mostly in the direction of the second axis; but once you clip it by value, you get [0.9,1.0], which points roughly in the diagonal between the two axes. 
In practice, this approach works well. If you want to ensure that Gradient Clipping does not change the direction of the gradient vector, you should clip by norm by setting `clipnorm` instead of `clipvalue`. This will clip the whole gradient if its ℓ norm is greater than the threshold you picked. For example, if you set clipnorm=1.0, then the vector [0.9, 100.0] will be clipped to [0.00899964, 0.9999595], preserving its orientation but almost
eliminating the first component.

**Comment by isaacgerg** [comment]
@sachinprasadhs You should add your commentary to the documentation.

 What does global_clipnorm do?

**Comment by sachinprasadhs** [comment]
As mentioned below also as per the document about `global_clipnorm`.
> @isaacgerg if clipnorm(float) is set, the gradient of each weight is individually clipped so that its norm is no higher than this value. If global_clipnorm (float) is set the gradient of all weights is clipped so that their global norm is no higher than this set value. This has been clearly mentioned in **kwargs.

**Comment by isaacgerg** [comment]
I don't see how clip_norm is different than global_clipnorm.  

Clip_ value examines each scalar and hard clips based on threshold but preserves sign.  Clip_norm rescales all the scalars so that the norm is the specified value.  But, what does global_clipnorm do?

**Comment by sachinprasadhs** [comment]
Does the description provided [here](https://www.tensorflow.org/api_docs/python/tf/clip_by_global_norm) helps you.

**Comment by isaacgerg** [comment]
I'm still here but its approaching the holidays in the USA.

**Comment by isaacgerg** [comment]
@sachinprasadhs Sort of.  Is the gradient of a model broken up into sections?  I presume it must otherwise, clip_by_global_norm will never take in a list.

**Comment by sachinprasadhs** [comment]
Note that gradient of NN is a list of Tensors, not scalars and in the code it internally calls `tf.clip_by_global_norm` check [here](https://github.com/keras-team/keras/blob/v2.7.0/keras/optimizer_v2/optimizer_v2.py#L428-L439) and [here](https://github.com/keras-team/keras/blob/2c48a3b38b6b6139be2da501982fd2f61d7d48fe/keras/optimizer_v2/utils.py#L106).
In future with the new release of keras optimizer it will be more easier for users and will be clear.
I will create a PR for the changes pointing to the `tf.clip_by_global_norm` for the users to understand better. Thanks!

**Comment by isaacgerg** [comment]
@ sachinprasadhs  This sounds great!  Thank you for your efforts!
